### PR TITLE
config.tf: bump images to latest available versions (incl. v1.6.6)

### LINKED ISSUE #1114: Update to Kubernetes/Bootkube/Hyperkube to v1.6.6

### PR DESCRIPTION
**Pull request by Quentin-M**
This PR musts cover::
- [x] hyperkube
- [x] bootkube
- [x] etcd + etcd operator
- [x] Ingress controller
- [x] KubeDNS
- [x] console (https://github.com/coreos/tectonic-installer/pull/1176)
- [x] CLUO update (https://github.com/coreos/tectonic-installer/pull/1178)
- [x] TCO (cc @yifan-gu)
- [x] Sync with upstream manifests

After the first RCs:
- [ ] KVO (cc @diegs @derekparker)
- [ ] Update Tectonic's payload (and verify TEO is included)

Fixes #1114.

### CODE REVIEW

**Comment by aaronlevy** [comment]
Hyperkube and dns bump lgtm

**Comment by discordianfish** [comment]
@Quentin-M Any reasons to wait for the other tectonic components before merging this? Given that vanilla k8s installation is supported it would make sense to update k8s independently of the tectonic bits, no? At least in more urgent updates like this.

**Comment by discordianfish** [comment]
Just tested this and it's working for me. But somehow the version it pulls has a -dirty tag:

```
xx-master0   Ready     12m       v1.6.6+coreos.0-dirty
xx-worker0   Ready     12m       v1.6.6+coreos.0-dirty
xx-worker1   Ready     12m       v1.6.6+coreos.0-dirty
xx-worker2   Ready     12m       v1.6.6+coreos.0-dirty
```

**Comment by aaronlevy** [comment]
I'll look into the '-dirty' build tag

**Comment by aaronlevy** [comment]
FWIW  the dirty working dir during the build process didn't have anything to do with the k8s code. But I'll need to resolve this in the jenkins build, then cut a new release.

**Comment by derekparker** [comment]
Version bumps for tectonic / kubernetes lgtm.

**Comment by aaronlevy** [comment]
There is now a `v1.6.6_coreos.1` hyperkube image that should be used (clean build).

**Comment by Quentin-M** [comment]
https://github.com/coreos/bootkube-terraform/pull/1/files can help determining what needs to change.

**Comment by Quentin-M** [comment]
Just used https://github.com/kubernetes-incubator/bootkube/compare/e0ab128640d513c8c88f9627b36ebfa94bcedc94...43be7e0bd6c47d1a6465984a4ecc82cedf83a770 in order to make sure our manifests are up to date with bootkube master (soon v0.4.5).

**Comment by Quentin-M** [comment]
ok to test

**Comment by yifan-gu** [comment]
@Quentin-M Also need to add `pod-checkpointer` to the bump list, waiting for a bootkube release so we can make an image for it.
/cc @aaronlevy

File: modules/bootkube/resources/experimental/bootstrap-manifests/bootstrap-etcd.yaml [hunk 1]
```diff
@@ -22,3 +22,4 @@ spec:
     - --data-dir=/var/etcd/data
   hostNetwork: true
   restartPolicy: Never
+  dnsPolicy: ClusterFirstWithHostNet
```
Comment (Quentin-M) [inline modules/bootkube/resources/experimental/bootstrap-manifests/bootstrap-etcd.yaml]: /cc @hongchaodeng @xiang90

Comment (xiang90) [inline modules/bootkube/resources/experimental/bootstrap-manifests/bootstrap-etcd.yaml]: lgtm.

**Comment by dghubble** [comment]
Updating to the new CLUO release is in https://github.com/coreos/tectonic-installer/pull/1178, along with the list of checks and verifications that were performed.

File: modules/bootkube/resources/experimental/bootstrap-manifests/bootstrap-etcd.yaml [hunk 2]
```diff
@@ -12,13 +12,31 @@ spec:
...
+    - --cert-file=/etc/kubernetes/secrets/operator/etcd-crt.pem
+    - --key-file=/etc/kubernetes/secrets/operator/etcd-key.pem
+    volumeMounts:
+    - mountPath: /etc/kubernetes/secrets
+      name: secrets
+      readOnly: true
   hostNetwork: true
+  dnsPolicy: ClusterFirstWithHostNet
   restartPolicy: Never
+  dnsPolicy: ClusterFirstWithHostNet
```
Comment (s-urbaniak) [inline modules/bootkube/resources/experimental/bootstrap-manifests/bootstrap-etcd.yaml]: I added the `dnsPolicy` in the original PR already, so this is added now twice here and in line 36. I suppose this doesn't matter, but we can remove the redundant entry.

Comment (Quentin-M) [inline modules/bootkube/resources/experimental/bootstrap-manifests/bootstrap-etcd.yaml]: Shoot probably at rebase 👎 Good catch!

**Pull request merged**